pkg/ruler/ruler.go

```diff
@@ -511,6 +516,7 @@ func (r *Ruler) syncManager(ctx context.Context, user string, groups store.RuleG
 		}
 		err = manager.Update(r.cfg.EvaluationInterval, files, nil)
 		if err != nil {
+			configUpdateFailuresTotal.WithLabelValues(user).Inc()
```

Comment:
Shouldn't we increment it also in case of `unable to create rule manager` few lines above?

Reply:
It's not clear to me if we should or shouldn't, the failure to create the ruler manager above seems less tied to the users config/rules files (which is what prompted this PR).

Reply:
If the above command fails that would signal a error that is a result of a bad Cortex config, not a user rules config. We should instrument that error, or ensure there is a mechanism to communicate a failure. But it probably should be a separate metric.

Reply:
A `reason` label should be an adequate solution